components/abstractions/src/main/java/com/microsoft/kiota/MultipartBody.java

```diff
@@ -48,19 +48,35 @@ public MultipartBody() {
      */
     public <T> void addOrReplacePart(
             @Nonnull final String name, @Nonnull final String contentType, @Nonnull final T value) {
+        addOrReplacePart(name, contentType, value, null);
+    }
+
+    /**
+     * Adds or replaces a part in the multipart body.
+     *
+     * @param <T> the type of the part to add or replace.
+     * @param name the name of the part to add or replace.
+     * @param contentType the content type of the part to add or replace.
+     * @param value the value of the part to add or replace.
+     * @param filename the value of the filename directive.
+     */
+    public <T> void addOrReplacePart(
+            @Nonnull final String name,
+            @Nonnull final String contentType,
+            @Nonnull final T value,
+            @Nullable String filename) {
         Objects.requireNonNull(value);
         if (Compatibility.isBlank(contentType))
             throw new IllegalArgumentException("contentType cannot be blank or empty");
         if (Compatibility.isBlank(name))
             throw new IllegalArgumentException("name cannot be blank or empty");
 
         final String normalizedName = normalizePartName(name);
-        originalNames.put(normalizedName, name);
-        parts.put(normalizedName, new AbstractMap.SimpleEntry<>(contentType, value));
+        Part part = new Part(name, value, contentType, filename);
```

Comment:
This line is failing the CI with the error 
```
Error: Access to private constructor of class Part requires synthetic accessor [SyntheticAccessor]

   Explanation for issues of type "SyntheticAccessor":
   A private inner class which is accessed from the outer class will force the
   compiler to insert a synthetic accessor; this means that you are causing
   extra overhead. This is not important in small projects, but is important
   for large apps running up against the 64K method handle limit, and
   especially for libraries where you want to make sure your library is as
   small as possible for the cases where your library is used in an app
   running up against the 64K limit.

```

Does it makes sense to change to using static a factory instead for the `Part` class?

Reply:
In this particular case, this class may not be static.

Reply:
Just to clarify, what I mean here is that we can add a static method to the type that returns a new instance of the class rather than a private constructor?

Reply:
@andrueastman maybe to avoid generating a synthetic constructor, we need to set it as package-scoped? That might be fair enough for Android.

Reply:
Yeah. That could also work. 
Any chance you can make the change so that we merge this in? @piotrooo

Reply:
Yeah, sure.